core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -154,6 +156,7 @@ func NewEthBroadcaster(
 		initSync:         sync.Mutex{},
 		isStarted:        false,
 		autoSyncNonce:    autoSyncNonce,
+		addrParser:       gethCommon.HexToAddress, // note: still evm-specific
```

Comment:
Oh this version is also trouble because it ignores both hex encoding errors and length mismatches.
WDYT about being proactive with an `error` return value on this func signature, and maybe at least checking `gethCommon.IsHexAddress` for now? 
```suggestion
		addrParser:       func(s string) (ADDR, error) {
			if !gtethCommon.IsHexAddress(s) {
				// err
			}
			return gethCommon.HexToAddress(s), nil
		}, // note: still evm-specific
```

Reply:
yeah i was wondering about how it handles errors with invalid strings - I'll add some error handling